src/filter/FilterByAmount.js

```diff
@@ -0,0 +1,39 @@
+import AmountType from '../types/AmountType';
+
+function FilterByAmount(resultAmount, filterMin, filterMax) {
+  const { min, max, type } = resultAmount;
+
+  if (type === AmountType.Range || type === AmountType.Fixed) {
+    // both ranges filled
+    if (min > 0 && max > 0)
+      return (
+        (filterMin > 0 &&
+          filterMax > 0 &&
+          min <= filterMax &&
+          max >= filterMin) ||
+        // only min fitler
+        (filterMin > 0 && filterMax === 0 && max >= filterMin) ||
+        // only max filter
+        (filterMin === 0 && filterMax > 0 && min <= filterMax)
+      );
+    // only min range filled
+    if (min > 0 && max === 0)
+      return (
+        filterMax >= min ||
+        // only min filter
+        filterMax === 0 ||
+        // only max filter
+        (filterMin === 0 && filterMax >= min)
+      );
+
+    // only max filled
+    if (max > 0 && min === 0) return filterMin <= max;
```

Comment:
I haven't dug too deeply on the logic here, but I don't understand why there's three cases here when we set `min` and `max` to reasonable defaults [`0` and `RANGE_MAX`](https://github.com/beyondhb1079/s4us/blob/afa48290e9f1d16397c3a9f1c5bd995e763ec8a0/src/types/ScholarshipAmount.ts#L53-L54) when they're not set that should allow for easier comparison.

I think this can simply be written as:
```js
      return (
        (filterMin > 0 &&
          filterMax > 0 &&
          min <= filterMax &&
          max >= filterMin) ||
        // only min fitler
        (filterMin > 0 && filterMax === 0 && max >= filterMin) ||
        // only max filter
        (filterMin === 0 && filterMax > 0 && min <= filterMax)
      );
```

Though that looks a bit messy, so I recommend writing it as this instead:

```js
if (filterMin && max < filterMin || filterMax && min > filterMax) {
  return false;
}
return true;
```

Yes, this could be written as `return !(filterMin && max < filterMin || filterMax && min > filterMax)` but I think this is more readable, at least since this is function instead of a lambda.

Also, considering the min/max [constants we use for FullTuition and Unknown amounts](https://github.com/beyondhb1079/s4us/blob/main/src/types/ScholarshipAmount.ts#L56-L63) I think this code works without needing to check the amount type too. If that's the case then this function is like max 3 lines long. I would inline it in a lambda where you're using it instead.